### PR TITLE
add Paella (Fast Text-Conditional Discrete Denoising on Vector-Quantized Latent Spaces) 

### PR DESCRIPTION
**Pull request by aengusng8**
Hi, @patrickvonplaten. _This is my **draft PR** that we recently mentioned in Discord. I am 90% complete to move Paella to our library, and I think I need your help to finalize this progress._

### **What is done?**
"Pipeline" and "Scheduler" are ready to run, check my Kaggle notebook: https://www.kaggle.com/code/aengusng/notebookd7ca68b633/notebook
Note: run this by CPU only in Colab, or CPU/GPU in Kaggle.

### **Current bottleneck problems?** 
I have a few questions that I would appreciate your help with:
1. Should I use layers, blocks, or models in the `diffusers\src\diffusers\models` folder to replace some parts of the original Paella model class, or should I keep the original Paella model class unchanged?
2. Can you please specify the file or folder I need to move the Paella model class into?
3. Can the code contain additional libraries such as `einops`, `rudalle`, and `open_clip_torch`, since they are part of the author's code?
4. When their `vqvae` is initialized from `rudalle.get_vae`, their `text_encoder` and `tokenizer` are initialized from `open_clip`, and How to save and upload the model class/configurations of `vqvae`, `text_encoder`, and `tokenizer` that **are outside of Diffusers** (like this https://huggingface.co/CompVis/stable-diffusion-v1-4)?

### **What is next?** 

1. Upon resolving this bottleneck, I plan to easily incorporate three additional pipelines: outpainting, image variation, and image interpolation.
2. Lastly, I will include some tests.

_updated: I apologize for the delay in completing this work due to the Lunar New Year. I am back and complete it asap._

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Hey @aengusng8,

Super cool! This already looks great :-) Please ping me @pcuenca if you'd like to have a review

File: src/diffusers/pipelines/paella/pipeline_paella.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+        x = self.embedding(x).permute(0, 3, 1, 2)
+        if len(c.shape) == 2:
+            s = torch.cat([c, r_embed], dim=-1)[:, :, None, None]
+        else:
+            r_embed = r_embed[:, :, None, None].expand(-1, -1, c.size(2), c.size(3))
+            s = torch.cat([c, r_embed], dim=1)
+        level_outputs = self._down_encode_(x, s)
+        x = self._up_decode(level_outputs, s)
+        x = self.clf(x)
+        return x
```
Comment (aengusng8) [inline src/diffusers/pipelines/paella/pipeline_paella.py:399]: Hi @pcuenca, I apologize for the delay in completing this work due to the Lunar New Year. I have a few questions that I would appreciate your help with:

1. Should I use layers, blocks, or models in the `diffusers\src\diffusers\models` folder to replace some parts of the original Paella model, or should I keep the original Paella model unchanged?
2. Can you please specify the file or folder I need to move the Paella model to (and send me the path example)?

Thank you in advance!

Comment (patrickvonplaten) [reply src/diffusers/pipelines/paella/pipeline_paella.py:399]: Hey @aengusng8,

No worries! Thanks a lot for working on this :-) 

It would be amazing if you could try to "mold" your code into the existing `UNet2DConditionModel` class: https://github.com/huggingface/diffusers/blob/2f9a70aa852d84adaa17a824454bf6d28180b55a/src/diffusers/models/unet_2d_condition.py#L53 given that Paella uses a text conditioned unet it should fit. 

Also we've just added a design philosophy that might help: https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy

So it be super cool if you could gauge whether it's possible to "force" the whole modeling code into UNet2DConditionModel - feel free to design your own, new unet up and down class

**Comment by dblunk88** [comment]
Any updates? 

Is this based on https://github.com/dome272/Paella ?

**Comment by aengusng8** [comment]
Hi @dblunk88,
1. No
2. Yes, but based on old Paella ([Paella](https://github.com/dome272/Paella) is recently updated)
